libsolidity/ast/Types.cpp

```diff
@@ -745,11 +745,13 @@ bool RationalNumberType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 {
 	if (_convertTo.category() == Category::Integer)
 	{
-		if (m_value == rational(0))
-			return true;
 		if (isFractional())
 			return false;
 		IntegerType const& targetType = dynamic_cast<IntegerType const&>(_convertTo);
+		if (targetType.isAddress())
```

Comment:
The problem with this architecture is that we cannot tell the reason why an implicit conversion did not work :(

Reply:
Perhaps we should add that to the caller of `isImplicitlyConvertibleTo` for the case of converting to address, otherwise the users are left in the dark.

Reply:
If it had access to pragmas, it could disallow only in V050 mode

Reply:
There should be access to pragmas. Have a grep in `Types.cpp` there must be something else using it already.

Reply:
Only code that I found with a quick find was a method that was passed a contract object.

Reply:
Is this a breaking change due to this problem?

Reply:
There doesn't seem to be a test case for this.

I think we could also just merge this PR without this single restriction. Perhaps for some cases it could be implemented in the type checker, but not sure it's worth it.

Reply:
So are you asking to allow 0 rational to address or all rational to address?

Reply:
Has this been answered? @axic

Reply:
I think this question is obsolete since we're merging into 0.5.0.